cmd/oci-catalog/src/providers/dockerhub.rs

```diff
@@ -137,6 +207,30 @@ fn url_for_request(request: &ListRepositoriesRequest) -> Url {
     url
 }
 
+fn url_for_request_tags(request: &ListTagsRequest) -> Result<Url, Status> {
+    let mut url = reqwest::Url::parse(DOCKERHUB_URI).unwrap();
+
+    let repo = match request.repository.clone() {
+        Some(r) => r,
+        None => {
+            return Err(Status::invalid_argument(format!(
+                "repository not set in request"
+            )))
+        }
+    };
+
+    url.set_path(&format!(
+        "/v2/repositories/{}/{}/tags",
+        repo.namespace, repo.name
+    ));
+
+    // For now we use a default page size.
+    url.query_pairs_mut()
+        .append_pair("page_size", &format!("{}", DEFAULT_PAGE_SIZE));
```

Comment:
Just to confirm, are we only fetching (`DEFAULT_PAGE_SIZE=100`) elements for now, aren't we? I mean, there I don't see any pagination logic here. It seems Docker is using a `Link` header to let the client know about the next page ([source](https://docs.docker.com/registry/spec/api/#pagination)), but we are not using it at the moment. No?

Reply:
That's just the current default pagesize for the requests we make to dockerhub (ie. how big are the batches that we request), so will affect performance only (number of requests required to exhaust the result). It's not limiting how many results we get at all (ie. if I set this to 10, then the loop will fetch the results in batches of 10).

The gRPC API that we provide streams the result back, so no pagination required.